proposals/zips/zip015.sol

```diff
@@ -0,0 +1,107 @@
+
+import {Addresses} from "@proposals/Addresses.sol";
+import {Proposal} from "@proposals/proposalTypes/Proposal.sol";
+import {TimelockProposal} from "@proposals/proposalTypes/TimelockProposal.sol";
+
+import {Core} from "@protocol/core/Core.sol";
+import {Roles} from "@protocol/core/Roles.sol";
+import {Token} from "@protocol/token/Token.sol";
+import {CoreRef} from "@protocol/refs/CoreRef.sol";
+import {ERC1155Sale} from "@protocol/sale/ERC1155Sale.sol";
+
+contract zip015 is TimelockProposal {
+    Core private _core;
+
+    constructor() Proposal("ADMIN_TIMELOCK_CONTROLLER") {}
+
+    // Returns the name of the proposal.
+    function name() public pure override returns (string memory) {
+        return "ZIP015";
+    }
+
+    // Provides a brief description of the proposal.
+    function description() public pure override returns (string memory) {
+        return "The ZTX marketplace sale conracts";
+    }
+
+    function _beforeDeploy() internal override {
+        _core = Core(addresses.getAddress("CORE"));
+    }
+
+    function _deploy() internal override {
+        /// ERC1155Sale
+        ERC1155Sale consumablesERC1155Sale = new ERC1155Sale(
+            address(_core),
+            addresses.getAddress("ERC1155_MAX_SUPPLY_MINTABLE_CONSUMABLES"),
+            addresses.getAddress("WETH")
+        );
+        addresses.addAddress("ERC1155_SALE_CONSUMABLES", address(consumablesERC1155Sale), true);
+
+        ERC1155Sale placeablesERC1155Sale = new ERC1155Sale(
+            address(_core),
+            addresses.getAddress("ERC1155_MAX_SUPPLY_MINTABLE_PLACEABLES"),
+            addresses.getAddress("WETH")
+        );
+        addresses.addAddress("ERC1155_SALE_PLACEABLES", address(placeablesERC1155Sale), true);
+
+        ERC1155Sale wearablesERC1155Sale = new ERC1155Sale(
+            address(_core),
+            addresses.getAddress("ERC1155_MAX_SUPPLY_MINTABLE_WEARABLES"),
+            addresses.getAddress("WETH")
+        );
+        addresses.addAddress("ERC1155_SALE_WEARABLES", address(wearablesERC1155Sale), true);
+
+        /// @dev configure the proceeds
+        ERC1155Sale(addresses.getAddress("ERC1155_SALE_CONSUMABLES")).setTokenRecipients(
+            addresses.getAddress("TOKEN"),
+            addresses.getAddress("REVENUE_WALLET_MULTISIG01"),
+            addresses.getAddress("REVENUE_WALLET_MULTISIG02")
+        );
+
+        ERC1155Sale(addresses.getAddress("ERC1155_SALE_PLACEABLES")).setTokenRecipients(
+            addresses.getAddress("TOKEN"),
+            addresses.getAddress("REVENUE_WALLET_MULTISIG01"),
+            addresses.getAddress("REVENUE_WALLET_MULTISIG02")
+        );
+
+        ERC1155Sale(addresses.getAddress("ERC1155_SALE_WEARABLES")).setTokenRecipients(
+            addresses.getAddress("TOKEN"),
+            addresses.getAddress("REVENUE_WALLET_MULTISIG01"),
+            addresses.getAddress("REVENUE_WALLET_MULTISIG02")
+        );
+    }
+
+    function _afterDeploy() internal override {
```

Comment:
These methods should be added in `_build()` after this PR https://github.com/ZTX-Foundation/tuxedo/pull/233

Reply:
Thanks @prateek105. Updated.